utils/autogenerate_all_imports.py

```diff
@@ -2,21 +2,48 @@
 """Pre-commit hook to add ``__all__`` imports to ``__init__`` files."""
 
 import ast
+import re
 import sys
 from collections import Counter
 from pathlib import Path
 from pprint import pprint
-from typing import Dict, List, Optional, Tuple
+from typing import Optional
 
+AUTO_GENERATED = 'AUTO-GENERATED'
+# Four-space indentation level (must be compatible with the formatter!)
+INDENT = '    '
 
-def parse_all(folder_path: str) -> Tuple[dict, dict]:
+
+def isort_alls(alls: set[str]) -> list[str]:
+    """Sort module according to 'isort style' as used by RUF022 rule
+
+    An isort-style sort orders items first according to their casing:
+    SCREAMING_SNAKE_CASE names (conventionally used for global constants)
+    come first, followed by CamelCase names (conventionally used for
+    classes), followed by anything else. Within each category,
+    a [natural sort](https://en.wikipedia.org/wiki/Natural_sort_order)
+    is used to order the elements.
+    """
+
+    constants = set(mod for mod in alls if re.match(r'^[A-Z][A-Z_]*$', mod))
+    # TODO: This doesn't represent cammel-case, only things that start with capital letters
+    # and are not screaming case.
+    classes = set(mod for mod in alls.difference(constants) if re.match(r'^[A-Z].*', mod))
+    others = alls.difference(classes, constants)
+    return sorted(constants) + sorted(classes) + sorted(others)
+
+
+def parse_all(folder_path: Path) -> tuple[dict, dict]:
     """Walk through all files in folder, and parse the ``__all__`` variable.
 
     :return: (all dict, dict of unparsable)
     """
-    folder_path = Path(folder_path)
-    all_dict = {}
-    bad_all = {}
+    all_dict: dict = {}
+    bad_all: dict[str, list] = {}
+
+    def is_ast_str(el: ast.AST) -> bool:
+        """Replacement for deprecated ast.Str"""
```

Comment:
`ast.Str` has been remove in Python 3.14

https://docs.python.org/3/library/ast.html#ast.Constant